### ISSUE TITLE
Getting "InstanceTaxonomies are not compatible for creating instance."

### ISSUE THREAD

**Issue by yaroslavvb** [GoogleCloudPlatform/hpc-toolkit#1568]
ghpc deploy fails to launch machine with `InstanceTaxonomies are not compatible for creating instance.`, how do I troubleshoot this?

- Step 1 [blueprint](https://www.dropbox.com/scl/fi/7pv9kzldjfi3jf7z092xu/dcluster-jun7.yaml?dl=0&rlkey=jufmxzlyyye8odeuz23me4xuw), simple cluster with `tiny` partition

- Step 2 [blueprint](https://www.dropbox.com/scl/fi/dkkhyvisuvihzvzrk2fn0/dcluster-jul11.yaml?dl=0&rlkey=x4ai4zgw1cfof6sw9l71lo408), add `c3` partition with `c3-standard-`4 node type, which fails to bring up machines with `InstanceTaxonomies are not compatible for creating instance` error

```
[2023-07-11T21:15:05.875] update_node: node dcluster1-c3-ghpc-0 state set to DOWN
[2023-07-11T21:15:05.875] update_node: node dcluster1-c3-ghpc-1 reason set to: [pd-standard] features and [instance_type: VIRTUAL_MACHINE
family: COMPUTE_OPTIMIZED
generation: GEN_3
cpu_vendor: INTEL
architecture: X86_64
] InstanceTaxonomies are not compatible for creating instance.
```

Here's the complete log

```
{
  "protoPayload": {
    "@type": "type.googleapis.com/google.cloud.audit.AuditLog",
    "status": {
      "code": 3,
      "message": "[pd-standard] features and [instance_type: VIRTUAL_MACHINE\nfamily: COMPUTE_OPTIMIZED\ngeneration: GEN_3\ncpu_vendor: INTEL\narchitecture: X86_64\n] InstanceTaxonomies are not compatible for creating instance."
    },
    "authenticationInfo": {
      "principalEmail": "1048622315445-compute@developer.gserviceaccount.com",
      "serviceAccountDelegationInfo": [
        {
          "firstPartyPrincipal": {
            "principalEmail": "service-1048622315445@compute-system.iam.gserviceaccount.com"
          }
        }
      ],
      "principalSubject": "serviceAccount:1048622315445-compute@developer.gserviceaccount.com"
    },
    "requestMetadata": {
      "callerIp": "34.142.203.74",
      "callerSuppliedUserAgent": "Slurm_GCP_Scripts/1.5 (GPN:SchedMD) (gzip),gzip(gfe)",
      "callerNetwork": "//compute.googleapis.com/projects/contextual-research-common/global/networks/__unknown__",
      "requestAttributes": {
        "time": "2023-07-11T21:03:06.725312Z",
        "auth": {}
      },
      "destinationAttributes": {}
    },
    "serviceName": "compute.googleapis.com",
    "methodName": "v1.compute.regionInstances.bulkInsert",
    "authorizationInfo": [
      {
        "permission": "compute.instances.create",
        "granted": true,
        "resourceAttributes": {
          "service": "compute",
          "name": "projects/contextual-research-common/zones/asia-southeast1-c/instances/unusedName",
          "type": "compute.instances"
        }
      },
      {
        "permission": "compute.disks.create",
        "granted": true,
        "resourceAttributes": {
          "service": "compute",
          "name": "projects/contextual-research-common/zones/asia-southeast1-c/disks/unusedName",
          "type": "compute.disks"
        }
      },
      {
        "permission": "compute.disks.setLabels",
        "granted": true,
        "resourceAttributes": {
          "service": "compute",
          "name": "projects/contextual-research-common/zones/asia-southeast1-c/disks/unusedName",
          "type": "compute.disks"
        }
      },
      {
        "permission": "compute.subnetworks.use",
        "granted": true,
        "resourceAttributes": {
          "service": "compute",
          "name": "projects/contextual-research-common/regions/asia-southeast1/subnetworks/contextual-a100-primary-subnet",
          "type": "compute.subnetworks"
        }
      },
      {
        "permission": "compute.instances.setMetadata",
        "granted": true,
        "resourceAttributes": {
          "service": "compute",
          "name": "projects/contextual-research-common/zones/asia-southeast1-c/instances/unusedName",
          "type": "compute.instances"
        }
      },
      {
        "permission": "compute.instances.setTags",
        "granted": true,
        "resourceAttributes": {
          "service": "compute",
          "name": "projects/contextual-research-common/zones/asia-southeast1-c/instances/unusedName",
          "type": "compute.instances"
        }
      },
      {
        "permission": "compute.instances.setLabels",
        "granted": true,
        "resourceAttributes": {
          "service": "compute",
          "name": "projects/contextual-research-common/zones/asia-southeast1-c/instances/unusedName",
          "type": "compute.instances"
        }
      },
      {
        "permission": "compute.instances.setServiceAccount",
        "granted": true,
        "resourceAttributes": {
          "service": "compute",
          "name": "projects/contextual-research-common/zones/asia-southeast1-c/instances/unusedName",
          "type": "compute.instances"
        }
      }
    ],
    "resourceName": "projects/contextual-research-common/regions/asia-southeast1/instances",
    "request": {
      "count": "2",
      "sourceInstanceTemplate": "https://www.googleapis.com/compute/v1/projects/contextual-research-common/global/instanceTemplates/dcluster1-compute-c3-ghpc-20230711210104409100000001",
      "instanceProperties": {
        "networkInterfaces": [
          {
            "subnetwork": "https://www.googleapis.com/compute/v1/projects/contextual-research-common/regions/asia-southeast1/subnetworks/contextual-a100-primary-subnet"
          }
        ],
        "disks": [
          {
            "type": "PERSISTENT",
            "mode": "READ_WRITE",
            "deviceName": "persistent-disk-0",
            "boot": true,
            "initializeParams": {
              "sourceImage": "projects/schedmd-slurm-public/global/images/family/slurm-gcp-5-7-hpc-centos-7",
              "diskSizeGb": "120",
              "diskType": "pd-standard",
              "labels": [
                {
                  "key": "ghpc_deployment",
                  "value": "dcluster1"
                },
                {
                  "key": "slurm_instance_role",
                  "value": "compute"
                },
                {
                  "key": "ghpc_blueprint",
                  "value": "dcluster"
                },
                {
                  "key": "ghpc_module",
                  "value": "schedmd-slurm-gcp-v5-node-group"
                },
                {
                  "key": "ghpc_role",
                  "value": "compute"
                },
                {
                  "key": "slurm_cluster_name",
                  "value": "dcluster1"
                }
              ]
            },
            "autoDelete": true,
            "interface": "SCSI"
          }
        ],
        "labels": [
          {
            "key": "ghpc_role",
            "value": "compute"
          },
          {
            "key": "ghpc_blueprint",
            "value": "dcluster"
          },
          {
            "key": "ghpc_deployment",
            "value": "dcluster1"
          },
          {
            "key": "ghpc_module",
            "value": "schedmd-slurm-gcp-v5-node-group"
          },
          {
            "key": "slurm_instance_role",
            "value": "compute"
          },
          {
            "key": "slurm_cluster_name",
            "value": "dcluster1"
          }
        ]
      },
      "locationPolicy": {
        "locations": [
          {
            "key": "zones/asia-southeast1-a",
            "value": {
              "preference": "DENY"
            }
          },
          {
            "key": "zones/asia-southeast1-b",
            "value": {
              "preference": "DENY"
            }
          }
        ],
        "targetShape": "ANY_SINGLE_ZONE"
      },
      "perInstanceProperties": [
        {
          "key": "dcluster1-c3-ghpc-1"
        },
        {
          "key": "dcluster1-c3-ghpc-0"
        }
      ],
      "@type": "type.googleapis.com/compute.regionInstances.bulkInsert"
    },
    "response": {
      "error": {
        "errors": [
          {
            "domain": "global",
            "reason": "badRequest",
            "message": "[pd-standard] features and [instance_type: VIRTUAL_MACHINE\nfamily: COMPUTE_OPTIMIZED\ngeneration: GEN_3\ncpu_vendor: INTEL\narchitecture: X86_64\n] InstanceTaxonomies are not compatible for creating instance."
          }
        ],
        "code": 400,
        "message": "[pd-standard] features and [instance_type: VIRTUAL_MACHINE\nfamily: COMPUTE_OPTIMIZED\ngeneration: GEN_3\ncpu_vendor: INTEL\narchitecture: X86_64\n] InstanceTaxonomies are not compatible for creating instance."
      },
      "@type": "type.googleapis.com/error"
    },
    "resourceLocation": {
      "currentLocations": [
        "asia-southeast1"
      ]
    }
  },
  "insertId": "bnmp07dhluy",
  "resource": {
    "type": "audited_resource",
    "labels": {
      "project_id": "contextual-research-common",
      "method": "compute.regionInstances.bulkInsert",
      "service": "compute.googleapis.com"
    }
  },
  "timestamp": "2023-07-11T21:03:04.635953Z",
  "severity": "ERROR",
  "logName": "projects/contextual-research-common/logs/cloudaudit.googleapis.com%2Factivity",
  "receiveTimestamp": "2023-07-11T21:03:07.345757898Z"
}
```

**Comment by nick-stroud** [comment]
`c3` machines do not support `pd-standard` disks, which is the default for `community/modules/compute/schedmd-slurm-gcp-v5-node-group` ([reference](https://cloud.google.com/compute/docs/general-purpose-machines#c3_series)).

You can fix this by adding the `disk_type: pd-ssd` to the settings of `id: c3_node_group`.

There is a change on develop that will allow for `pd-balanced` and `pd-extreme` also to work, but this change is not yet released and will only be available in Toolkit versions >v1.20.x.

**Comment by nick-stroud** [comment]
I have also added validations to provide a clearer error message in #1579.

These validations are on develop now and will be in the next released version of the HPC Toolkit.
